components/bookshelf/discover/SearchBox.tsx

```diff
@@ -33,17 +30,23 @@ const SearchBox = ({ setIsSearching }: SearchBoxProps) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
-  const handleChange = (e) => {
+  const handleChange = (e: {
+    target: { value: React.SetStateAction<string> }
+  }) => {
     setSearchInput(e.target.value)
```

Comment:
Can you try if this works 

```
    setSearchInput(e.currentTarget.value)
```

Reply:
this one doesn't work

Reply:
Hmmm 🤔 

try this

```
  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setSearchInput(e.target.value)
  }
```

Anyway - the reason for this confusion is because tailwind-styled-components' `typescript` autocomplete is disabled. Which is why if you hover above `SearchInput`, you will see `any` - giving you no indication what it requires

![image](https://user-images.githubusercontent.com/780659/191931311-ec0ed5b9-2846-4946-8f43-212babcdf00a.png)

Reply:
![image](https://user-images.githubusercontent.com/91857581/191932558-7af3f193-160e-4d37-84a7-a4cc817017ee.png)

it works! nice

Reply:
this one also works 
![image](https://user-images.githubusercontent.com/91857581/191933009-b78f7fee-de11-4cdd-8e83-7075b6448ed7.png)

Reply:
> Anyway - the reason for this confusion is because tailwind-styled-components' `typescript` autocomplete is disabled. Which is why if you hover above `SearchInput`, you will see `any` - giving you no indication what it requires

checked my settings.json and `"typescriptreact": "javascript"` is already set there. am I missing something ba? 
![image](https://user-images.githubusercontent.com/91857581/191933196-1c8a9b76-ef1a-4fc1-94ed-969a4337bc76.png)

or maybe I have to add both?
![image](https://user-images.githubusercontent.com/91857581/191933500-fdc5ec22-2096-4588-865f-d5f51d0b4527.png)